dm/dm/worker/server.go

```diff
@@ -89,12 +89,13 @@ func NewServer(cfg *Config) *Server {
 	return &s
 }
 
-// Start starts to serving.
+// Start starts to serving. It will block current goroutine until Server.Close or
+// input context canceled.
 // this function should only exit when can't dail DM-master, for other errors it should not exit.
-func (s *Server) Start() error {
+func (s *Server) Start(ctx context.Context) error {
```

Comment:
why not make start/close mutual exclusive?

Reply:
If so, when Server.Start is stuck, we can't Close it.

Reply:
if we reached `m.Serve()`, `cancel()` cannot cancel it, we need `close` which close the listener.

we only need to make sure field init part(initializing) mutual exclusive with close